api/clients/retrieval_client_v2.go

```diff
@@ -0,0 +1,204 @@
+package clients
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	grpcnode "github.com/Layr-Labs/eigenda/api/grpc/node/v2"
+	"github.com/Layr-Labs/eigenda/core"
+	corev2 "github.com/Layr-Labs/eigenda/core/v2"
+	"github.com/Layr-Labs/eigenda/encoding"
+	"github.com/Layr-Labs/eigensdk-go/logging"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+
+	"github.com/gammazero/workerpool"
+)
+
+// RetrievalClientV2 is an object that can retrieve blobs from the network.
+type RetrievalClientV2 interface {
+	// GetBlob downloads chunks of a blob from operator network and reconstructs the blob.
+	GetBlob(ctx context.Context, blobHeader corev2.BlobHeader, referenceBlockNumber uint64, quorumID core.QuorumID) ([]byte, error)
+}
+
+type retrievalClientV2 struct {
+	logger            logging.Logger
+	ethClient         core.Reader
+	indexedChainState core.IndexedChainState
+	verifier          encoding.Verifier
+	numConnections    int
+}
+
+// NewRetrievalClientV2 creates a new retrieval client.
+func NewRetrievalClientV2(
```

Comment:
This fetches the data directly from the DA nodes. Should we first attempt to fetch the data from the relays? Or is this the job of another layer of the system?

Reply:
> This fetches the data directly from the DA nodes. Should we first attempt to fetch the data from the relays? Or is this the job of another layer of the system?

Yes, that is done using the relay client. This client is meant to be the client used to retrieve from the DA nodes. 
Whatever is integrating retrieval should choose to integrate using one or both of these clients.

Reply:
Clarified this in the comment